proof/crefine/AARCH64/Retype_C.thy

```diff
@@ -3135,7 +3137,8 @@ lemma cnc_tcb_helper:
                                      [heap_update (registers_of_tcb_Ptr p)
                                                   (array_updates (h_val (hrs_mem hrs) (registers_of_tcb_Ptr p))
                                                                  initContext_registers),
-                                      heap_update (machine_word_Ptr &(p\<rightarrow>[''tcbTimeSlice_C''])) 5])
+                                      heap_update (machine_word_Ptr &(p\<rightarrow>[''tcbTimeSlice_C''])) 5,
+                                      heap_update (machine_word_Ptr &(p\<rightarrow>[''tcbDomain_C''])) (d' :: machine_word)])
```

Comment:
I'm a bit confused here... this looks like the C changed somehow, but we're only changing the det_ext definitions on the abstract level, right? So why is this change needed all the way down in CRefine?

Reply:
I'd have to spend some time digging through the proofs to say for certain but I think this is due to the restructuring of `createNewCaps` in the exec spec, which was simplified to do everything for tcbs in one step, instead of setting the domains in a separate step. This is then reflected into how these helper lemmas in CRefine split off commands.

Reply:
That sounds fair, even if initially unexpected